File: devtrouble-ai-frontend/src/types/project.ts
export interface Project {
  id: string;
  owner_id: string | null;
  name: string;
  description: string | null;
  created_at: string;
  updated_at: string;
}

export interface ProjectCreateRequest {
  name: string;
  description?: string;
}

export type ProjectUpdateRequest = Partial<ProjectCreateRequest>;

File: devtrouble-ai-backend/app/schemas/project.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ProjectCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    description: str | None = Field(default=None, max_length=500)


class ProjectUpdateRequest(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=100)
    description: str | None = Field(default=None, max_length=500)


class ProjectResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    owner_id: str | None
    name: str
    description: str | None
    created_at: datetime
    updated_at: datetime
